common/djangoapps/enrollment/tests/test_views.py

```diff
@@ -922,13 +924,16 @@ def test_update_enrollment_with_expired_mode_throws_error(self):
         mode.expiration_datetime = datetime.datetime(year=1970, month=1, day=1, tzinfo=pytz.utc)
         mode.save()
         self.assert_enrollment_status(
-            as_server=True,
+            as_server=server_call,
             mode=CourseMode.VERIFIED,
-            expected_status=status.HTTP_400_BAD_REQUEST
+            expected_status=status.HTTP_200_OK if server_call else status.HTTP_400_BAD_REQUEST
         )
         course_mode, is_active = CourseEnrollment.enrollment_mode_for_user(self.user, self.course.id)
         self.assertTrue(is_active)
-        self.assertEqual(course_mode, CourseMode.DEFAULT_MODE_SLUG)
+        if server_call:
```

Comment:
I don't care much for if-statements in tests. This would be cleaner if you passed the expected course mode as an argument via DDT.